lib/storage/storage.ts

```diff
@@ -1,75 +1,11 @@
-import { BadRequestError, InternalServerError } from 'errors/apps-sdk-error';
-import { RequestOptions } from 'types/fetch';
+import {BaseStorage} from 'lib/storage/base-storage';
 import { isDefined } from 'types/guards';
-import { ErrorResponse, IStorageInstance, Options, SetResponse, Token } from 'types/storage';
-import { fetchWrapper } from 'utils/fetch-wrapper';
-import { Logger } from 'utils/logger';
+import { ErrorResponse, IStorageInstance, Options, SetResponse } from 'types/storage';
 
-const logger = new Logger('Storage', { mondayInternal: true });
-
-const getStorageUrl = () => {
-  const url = process.env.STORAGE_URL || 'https://apps-storage.monday.com/app_storage_api/v2';
-  return url;
-};
-
-const getToken = (token?: Token, options: Options = {}): Token => {
-  const selectedToken = options.token || token;
-
-  if (!isDefined(selectedToken)) {
-    throw new BadRequestError('Missing token');
-  }
-
-  return selectedToken;
-};
-
-const generateCrudPath = (key: string, options: Options) => {
-  if (!isDefined(key)) {
-    throw new BadRequestError('Missing key');
-  }
-
-  const { shared } = options;
-  const storageUrl = getStorageUrl();
-  // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-  const fullPath = `${storageUrl}/${key}?shareGlobally=${!!shared}`;
-  return fullPath;
-};
-
-const storageFetch = async <T>(key: string, initToken: Token | undefined, externalOptions: Options, options: RequestOptions) => {
-  const { method, body } = options;
-  const token = getToken(initToken, externalOptions);
-  const stringifiedBody = JSON.stringify(body);
-  const url = generateCrudPath(key, externalOptions);
-  if (!isDefined(method)) {
-    throw new InternalServerError('An error occurred');
-  }
-
-  const headers = {
-    Authorization: token,
-    'Content-Type': 'application/json'
-  };
-
-  let response: T | undefined;
-  try {
-    response = await fetchWrapper<T>(url, {
-      method,
-      headers,
-      ...(body && { body: stringifiedBody })
-    });
-  } catch (error: unknown) {
-    logger.debug('[storageFetch] Unexpected error occurred while communicating with storage', { error: error as Error });
-    throw new InternalServerError('An issue occurred while accessing storage');
-  }
-
-  return response as T;
-};
-
-export class Storage implements IStorageInstance {
-  constructor(private token?: Token) {
-  }
+export class Storage extends BaseStorage implements IStorageInstance {
```

Comment:
`this.token` is used in each CRUD method in this concrete class, because the base class `storageFetch` method requires it.
even if the token is being passed to the constructor of the concrete class, it is a common member through the base class, in which case, the base method `storageFetch` implementation can itself use `this.token`

Reply:
moreover, the base class under the hood selects the token with the `getToken()` method. as it is a private concrete method of the base class, it does not require the `token?: Token` param, as it can check the "local" class member `this.token`. so should only rely on "externalOptions" to be passed from outside.

wdyt?